argocd/schema_project.go

```diff
@@ -576,6 +576,11 @@ func projectSpecSchemaV2() *schema.Schema {
 					Required: true,
 					Elem:     &schema.Schema{Type: schema.TypeString},
 				},
+				"source_namespaces": {
```

Comment:
While this works, and I am arriving very very late to the party (sorry!), I wonder if `TypeSet` of Strings would not have been preferable here so that a change in the list elements' ordering will not retrigger a resource update, but it changes a lot of the logic underneath, and would imply some additional work. Therefore, I'm fine with letting it as is.
Look at how `jq_path_expressions` is handled for reference.

Reply:
This is a great point. I took a look and I don't think it will take much to rework this implementation so I'll get on it.